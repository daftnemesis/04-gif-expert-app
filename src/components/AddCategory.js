import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

  //agregando estado para el input para la busqueda
  const [inputValue, setInputValue] = useState('')

  //handler para manejar el evento del input
  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  //handler para manejar el evento de enter y no refrescar la pagina
  //previniendo el comportamiento por defecto
  const handleSubmit = (e) => {
    e.preventDefault()

    if(inputValue.trim().length > 2) {
      setCategories(cats => [inputValue, ...cats])
      setInputValue('')
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <p> {inputValue}</p>
      <input 
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

//Agregando proptypes para hacer requerido el setCategories
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}