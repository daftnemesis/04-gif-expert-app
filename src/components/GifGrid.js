import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

  const [images, setImages] = useState([])

  // el hook useEffect se usa para hacer cambios en el estado pero solo cuando el componente es renderizado
  useEffect( () => {
    getGifs();
  }, [])

  // Metodo para usar la API de giphy, usando un parametro predeterminado
  const getGifs = async () => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=scooby &limit=10&api_key=jnfYUZIxm7WR1pG7mWAcxrSQ7pGkJv5h'; // url de la peticion
    const resp = await fetch(url); //se usa el fetch
    const {data} = await resp.json(); // se pasa a formato json para poder trabajar con el

    // se usa el map para llebar la peticion a un arreglo y solo con las propiedades que se necesitan
    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images.downsized_medium.url,
      }
    })

    console.log(gifs);

    setImages(gifs) // se llena el arreglo con los gifs retornados de la peticion
  }

  return (
    <>
      <h3>{category}</h3>
      <div className='card-grid'>
        <ol>
          {
            images.map( img => (
              <GifGridItem 
                key={img.id}
                {...img}
              />
            ))
          }
        </ol>
      </div>
    </>
  )
}
