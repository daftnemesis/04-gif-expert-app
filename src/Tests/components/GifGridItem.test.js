import React from 'react'
import {shallow} from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => { 

  const url = 'https://google.com'
  const title = 'hola mundo'

  const wrapper = shallow(<GifGridItem url={url} title={title}/>)


  test('debe de mostrar el componente correctamente', () => {
    

    expect(wrapper).toMatchSnapshot()

  });

  test('deve de tener un parrafo con el title', () => {

    const p = wrapper.find('p')

    expect(p.text().trim()).toBe(title)
    
  });

  test('debe de tener la imagen ifual al url y alt de los props', () => {

    const img = wrapper.find('img')

    // console.log(img.props('src'))

    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
    
  });

  test('deve de tener animate__fadeIn', () => {

    const div = wrapper.find('div')

    expect(div.hasClass('animate__fadeIn')).toBe(true)
    
  });

})