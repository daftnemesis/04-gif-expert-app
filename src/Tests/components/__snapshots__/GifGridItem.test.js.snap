// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="hola mundo"
    src="https://google.com"
  />
  <p>
    hola mundo
  </p>
</div>
`;
